graphdatascience/query_runner/neo4j_query_runner.py

```diff
@@ -17,3 +18,7 @@ def run_query(self, query: str, params: Dict[str, Any] = {}) -> QueryResult:
 
     def set_database(self, db: str) -> None:
         self._db = db
+
+    def __del__(self) -> None:
+        if self._auto_close:
+            self._driver.close()
```

Comment:
According to the docs, closing the driver also closes all open connections/sessions. I assume the only way of checking if it's safe to close the driver is to know if there are still open sessions. 

There seems to be a field on the `Driver` https://github.com/neo4j/neo4j-python-driver/blob/3389c4ad0f51583ebd6ceaf903e7555f45d9a609/neo4j/_sync/driver.py#L296-L300 which we could use in a test? OTOH, I don't see any tests in the driver impl for testing close functionality, so I guess this is fine.

Reply:
This is very interesting. I'm using the latest released version of the driver (4.4.2) and there is no `Driver._close` field in my installation (see https://github.com/neo4j/neo4j-python-driver/blob/a0263b05048dbb379f8f4989842d0785cbde7c86/neo4j/__init__.py#L268).
I guess we can test this more easily in the 5.0 driver. In fact, then we can look at whether we get warnings or not (https://github.com/neo4j/neo4j-python-driver/blob/3389c4ad0f51583ebd6ceaf903e7555f45d9a609/neo4j/_sync/driver.py#L263) and we'd not have to rely on private internal fields in the driver.